### PR TITLE
CDI-578 reword original proposal.

### PR DESCRIPTION
**Pull request by tremes**
I tried to reword the original PR from Mark to include the original suggestions.

### CODE REVIEW

File: api/src/main/java/javax/enterprise/event/TransactionPhase.java [hunk 1]
```diff
@@ -21,13 +21,16 @@
  * Distinguishes the various kinds of transactional {@linkplain javax.enterprise.event.Observes observer methods} from regular
  * observer methods which are notified immediately.
  * </p>
- * 
  * <p>
  * Transactional observer methods are observer methods which receive event notifications during the before or after completion
  * phase of the transaction in which the event was fired. If no transaction is in progress when the event is fired, they are
  * notified at the same time as other observers.
+ * If the transaction is in progress, but {@link javax.transaction.Synchronization} callback cannot be registered due to the transaction is already
```
Comment (manovotn) [inline api/src/main/java/javax/enterprise/event/TransactionPhase.java]: I would change the sentence a bit (`due to the transaction is` sounds weird). How about this:
`If the transaction is in progress, but {@link javax.transaction.Synchronization} callback cannot be registered due to the transaction being already marked for rollback or in state where {@link javax.transaction.Synchronization} callbacks cannot be registered, ... `

Comment (tremes) [inline api/src/main/java/javax/enterprise/event/TransactionPhase.java]: Agree that it sounds weird. I think we can go with your proposal.

File: api/src/main/java/javax/enterprise/event/TransactionPhase.java [hunk 2]
```diff
@@ -45,13 +48,23 @@
      * <p>
      * Identifies a before completion observer method, called during the before completion phase of the transaction.
      * </p>
+     * <p>
+     * It will also get invoked if there is no transaction in progress, or the transaction is in progress,
```
Comment (mkouba) [inline api/src/main/java/javax/enterprise/event/TransactionPhase.java]: I'm not a native speaker but "get invoked" sounds wrong.

Comment (manovotn) [inline api/src/main/java/javax/enterprise/event/TransactionPhase.java]: I'd say it's OK as long as we talk about methods (which we do - it's OM here).
But we might also use something like "get notified" if that sounds better?

**Comment by mkouba** [comment]
Looks good.

**Comment by manovotn** [comment]
+1

**Pull request merged**